src/main/java/core/basesyntax/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+import java.util.Map;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    @Override
+    public String getReport(Map<String, Integer> storage) {
```

Comment:
The method `getReport` correctly generates a report based on the given storage map. Just ensure that you're not violating the encapsulation principle by modifying the `storage` map from outside this method.